jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/ClientUpgradeRequest.java

```diff
@@ -187,6 +189,10 @@ public void setSubProtocols(List<String> protocols)
     @Override
     public void send(final Response.CompleteListener listener)
     {
+        frameHandler = getFrameHandler(wsClient);
+        if (frameHandler == null)
+            throw new WebSocketException("FrameHandler was null");
```

Comment:
You cannot throw from here (it's all async), you have to call `abort(new WebSocketException())` and return.